metrics-core/src/main/java/com/yammer/metrics/core/PollingGauge.java

```diff
@@ -17,14 +17,14 @@ private PollingGauge(final Gauge<T> delegate, final Histogram histogram, Schedul
     this.future = pollThread.scheduleAtFixedRate(new Runnable() {
       @Override
       public void run() {
-        final long value;
+        final Number value;
         try {
-          value = delegate.value().longValue();
+          value = delegate.value();
         } catch (Throwable ignored) {
           return;
         }
 
-        histogram.update(value);
+        histogram.update(value.doubleValue());
```

Comment:
the `int`/`long` versions just call the `double` version, so I think we can just update this code to always convert to double:
https://github.com/HubSpot/metrics/blob/9d9ba277fb558fbdb62ee30cc0d43241e201ea8b/metrics-core/src/main/java/com/yammer/metrics/core/Histogram.java#L105-L116